src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/LocalTypeSpecSdkGenerationService.cs

```diff
@@ -0,0 +1,286 @@
+            Logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            ProcessExecutor = processExecutor ?? throw new ArgumentNullException(nameof(processExecutor));
+            TypeSpecSourcePath = typeSpecSourcePath ?? throw new ArgumentNullException(nameof(typeSpecSourcePath));
+            SdkOutputDirectory = sdkOutputDirectory ?? throw new ArgumentNullException(nameof(sdkOutputDirectory));
+            if (string.IsNullOrWhiteSpace(TypeSpecSourcePath))
+            {
+                throw new ArgumentException("TypeSpec source path cannot be null or whitespace", nameof(typeSpecSourcePath));
+            }
+
+            if (string.IsNullOrWhiteSpace(SdkOutputDirectory))
+            {
+                throw new ArgumentException("SDK output directory cannot be null or whitespace", nameof(sdkOutputDirectory));
+            }
+        }
+
+        protected virtual bool DirectoryExists(string path) => Directory.Exists(path);
+        protected virtual bool FileExists(string path) => File.Exists(path);
+        protected virtual void CreateDirectory(string path) => Directory.CreateDirectory(path);
+        protected virtual void DeleteDirectory(string path, bool recursive) => Directory.Delete(path, recursive);
+        protected virtual string[] GetFiles(string path) => Directory.GetFiles(path);
+        protected virtual string[] GetDirectories(string path) => Directory.GetDirectories(path);
+        protected virtual void MoveFile(string sourceFileName, string destFileName) => File.Move(sourceFileName, destFileName);
+        protected virtual void ReplaceFile(string sourceFileName, string destinationFileName, string destinationBackupFileName) => File.Replace(sourceFileName, destinationFileName, destinationBackupFileName);
+        protected virtual void DeleteFile(string path) => File.Delete(path);
+        protected virtual void MoveDirectory(string sourceDirName, string destDirName) => Directory.Move(sourceDirName, destDirName);
+
+        /// <summary>
+        /// Compiles a TypeSpec project into an SDK using the new simplified flow.
+        /// </summary>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <returns>True if compilation succeeds, false otherwise</returns>
+        public async Task<bool> CompileTypeSpecAsync(CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                if (!DirectoryExists(TypeSpecSourcePath))
+                {
+                    throw new DirectoryNotFoundException($"TypeSpec project directory not found: {TypeSpecSourcePath}");
+                }
+
+                Logger.LogInformation("Starting TypeSpec compilation for project: {ProjectPath}", TypeSpecSourcePath);
+                Logger.LogInformation("Output SDK path: {OutputPath}", SdkOutputDirectory);
+
+                if (!await InstallTypeSpecDependencies(cancellationToken))
```

Comment:
Just a small note on style - typically methods that indicate success or failure via a bool result would be called `Try*`